src/engines/html5/media-source/base-media-source-adapter.js

```diff
@@ -235,6 +237,22 @@ export default class BaseMediaSourceAdapter extends FakeEventTarget implements I
     this._videoElement.addEventListener(Html5EventType.DURATION_CHANGE, this._onDurationChanged);
   }
 
+  /**
+   * Disables all the existing text tracks.
+   * @public
+   * @returns {void}
+   */
+  disableNativeTextTracks(): void {
+    let textTracks = this._videoElement.textTracks;
+    if (textTracks) {
+      for (let i = 0; i < textTracks.length; i++) {
+        (textTracks[i].kind === 'subtitles' || textTracks[i].kind === 'captions') &&
+          (textTracks[i].language !== EXTERNAL_TRACK_ID || textTracks[i].label !== EXTERNAL_TRACK_ID) &&
+          (textTracks[i].mode = 'disabled');
```

Comment:
wouldn't an if statement make this condition more readable?
```javascript
        if (['subtitles', 'captions'].includes(textTracks[i].kind) && (
          textTracks[i].language !== EXTERNAL_TRACK_ID || textTracks[i].label !== EXTERNAL_TRACK_ID
        )) {
          textTracks[i].mode = 'disabled';
        }
```

we can also probably reduce the EXTERNAL_TRACK_ID to use find or some methods

Reply:
simplified @OrenMe  @yairans